treeview/js/main.js

```diff
@@ -13,16 +15,22 @@
             }
             return s4() + s4() + s4() + s4() + s4() + s4();
         }
-
         function getHeader(action, schemaTable, params) {
-            return ERMrest._certifyContextHeader({
+            var header={
                 wid: window.name,
                 pid: uuid(),
                 cid: "treeview",
                 action: action,
                 schema_table: schemaTable,
-                params: params
-            });
+                params: params,
+            }
+            if(action==="main"){
```

Comment:
If this change is only for one request, it's better to add the pcid and ppid right before that request is sent instead of adding a conditional statement here.

Reply:
Removed the code from getHeader() function and added the pcid and ppid attributes to main request and added 
ERMrest.certifyHeaders() to individual requests.